workflow/rules/activity_only_features.smk

```diff
@@ -19,18 +17,31 @@ rule activity_only_features:
 	script:
 		"../scripts/activity_only_features.R"
 
-rule gen_final_features:
-	# We really only do a file renaming here
-	# This step should be replaced by other modules if they wish to add more features
+# add external features (modify this rule in future pipelines- note, outputs should have name INPUT COL and will be renamed later)
+rule add_external_features:
 	input:
-		predictions_extended = os.path.join(RESULTS_DIR, "{biosample}", "ActivityOnly_features.tsv.gz"),
+		predictions_extended = os.path.join(RESULTS_DIR, "{biosample}", "ActivityOnly_features.tsv.gz")
 	output:
-		final_features = os.path.join(RESULTS_DIR, "{biosample}", "final_features.tsv.gz")
+		plus_external_features = os.path.join(RESULTS_DIR, "{biosample}", "ActivityOnly_external_features.tsv.gz")
 	conda:
 		"../envs/encode_re2g.yml"
 	resources:
 		mem_mb=1*1000
 	shell:
 		"""
-		cp {input.predictions_extended} {output.final_features}
-		"""
\ No newline at end of file
+		mv {input.predictions_extended} {output.plus_external_features}
+		"""
+
+# compute interaction or squared terms, fill NAs, rename features to finals, fill nas
+rule gen_final_features:
+	input:
+		plus_external_features = os.path.join(RESULTS_DIR, "{biosample}", "ActivityOnly_external_features.tsv.gz"),
+		feature_table_file = lambda wildcards: get_feature_table_file(wildcards.biosample)
+	output:
+		final_features = os.path.join(RESULTS_DIR, "{biosample}", "final_features.tsv.gz")
+	conda:
+		"../envs/encode_re2g.yml"
+	resources:
+		mem_mb=32*1000
```

Comment:
Using this fn tries to estimate the memory needed based on the input size and has built in retries that increase memory resources. We should try to use it unless the memory usage somehow doesn't scale linearly with the input size

```suggestion
		mem_mb=determine_mem_mb
```